### DISCUSSION TITLE
Status of Avalonia.Bionic?

### DISCUSSION THREAD

**Discussion by jhimes144** [AvaloniaUI/Avalonia#16798] [Q&A] [answered]
Hi!

Can anyone shine light on the status of Avalonia.Bionic? I'm super excited to hear about a huge increase in android performance and haven't seen a lot of news on the subject.

**Comment by 996666925** [comment]
I also want to know

**Comment by Taiizor** [comment]
Yes, many people want to know this.

**Comment by DoubleDBE** [comment]
I'm also curious.

**Comment by MikeCodesDotNET** [comment, marked as answer]
We're still evaluating Avalonia.Bionic's future. 

While we've learned valuable lessons that will improve our XPF Android performance, we're cautious about fully developing it for use with Avalonia. This caution stems from the possibility of Microsoft making significant improvements, potentially rendering our work redundant. We're carefully weighing the investment required against the potential benefits and risks. 

If we do decide to release Avalonia.Bionic, it would be as a paid feature to justify the development costs. However, no final decisions have been made yet, and the only concrete plan is to enable it for XPF customers.

**Reply by jhimes144** [reply]
Thanks Mike!

At the end of the day Avalonia has bills to pay, and fixing Microsoft's issues is not something anyone wants to tackle on a whim.

I see that MAUI is plagued by similar issues https://github.com/dotnet/maui/issues/18838

Could you speak to what piece is shared between Avalonia's Android and MAUI's (if any)? Do both Avalonia and MAUI share the same blight? If they do, I also would be willing to bet that Microsoft will fix this. It seems very critical as their focus is mobile. But, given the controversial state of MAUI - who knows! It may even come down to them either fixing this issue or discontinuing MAUI.

**Reply by anose001** [reply]
@MikeCodesDotNET , revisiting this thread and wondering whether Avalonia Bionic is planned for 2026 sometime?

**Reply by maxkatz6** [reply]
.NET 10 introduced experimental NativeAOT and CoreCLR support for Android. With stabilization on their side, it makes new target a first-class citizen target of .NET SDK itself. Without relying on slower MONO.

The same target can be used with MAUI, as it's done on SDK level.

**Reply by anose001** [reply]
Thanks for the quick response and the extra context on .NET 10! That's helpful
